pkg/metrics/gatherer/optsRegistry.go

```diff
@@ -0,0 +1,283 @@
+	// DurationAPIMetricName is the variable that stores the metric name for the duration API details.
+	DurationAPIMetricName = "cluster_installation_duration_api"
+
+	// MODIFICATION METRICS
+
+	// ModificationConfigMetricName is the variable that stores the metric name for the modification of config manifests details.
+	ModificationConfigMetricName = "cluster_installation_modification_config_manifest"
+	// ModificationDNSMetricName is the variable that stores the metric name for the modification of DNS manifests details.
+	ModificationDNSMetricName = "cluster_installation_modification_dns_manifest"
+	// ModificationNetworkMetricName is the variable that stores the metric name for the modification of network manifests details.
+	ModificationNetworkMetricName = "cluster_installation_modification_network_manifest"
+	// ModificationSchedulerMetricName is the variable that stores the metric name for the modification of scheduler manifests details.
+	ModificationSchedulerMetricName = "cluster_installation_modification_scheduler_manifest"
+	// ModificationCVOMetricName is the variable that stores the metric name for the modification of CVO manifests details.
+	ModificationCVOMetricName = "cluster_installation_modification_cvo_manifest"
+	// ModificationEtcdMetricName is the variable that stores the metric name for the modification of etcd manifests details.
+	ModificationEtcdMetricName = "cluster_installation_modification_etcd_manifest"
+	// ModificationMachineConfigMetricName is the variable that stores the metric name for the modification of machine config manifests details.
+	ModificationMachineConfigMetricName = "cluster_installation_modification_machineconfig_manifest"
+	// ModificationCaMetricName is the variable that stores the metric name for the modification of ca manifests details.
+	ModificationCaMetricName = "cluster_installation_modification_ca_manifest"
+	// ModificationPullSecretMetricName is the variable that stores the metric name for the modification of pull secret manifests details.
+	ModificationPullSecretMetricName = "cluster_installation_modification_pullsecret_manifest"
+	// ModificationBootstrapMetricName is the variable that stores the metric name for the modification of bootstrap manifests details.
+	ModificationBootstrapMetricName = "cluster_installation_modification_bootstrap_ignition"
+	// ModificationMasterMetricName is the variable that stores the metric name for the modification of master manifests details.
+	ModificationMasterMetricName = "cluster_installation_modification_master_ignition"
+	// ModificationWorkerMetricName is the variable that stores the metric name for the modification of worker manifests details.
+	ModificationWorkerMetricName = "cluster_installation_modification_worker_manifest"
+)
+
+var (
+	// optsRegistry holds all the information that every metric needs to create a metric builder.
+	optsRegistry = map[string]*builder.MetricOpts{
+		CreateMetricName: &builder.MetricOpts{
```

Comment:
I think it makes sense to initialize metrics with all the label values that are currently available, at least OS should be available. Not sure about any others.

Reply:
I added a utility function that sets the known information when the invocation metrics is actually needed. Placing it here seemed like we would populate metric values for metrics we may not need send out anyway.